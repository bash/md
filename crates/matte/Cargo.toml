[package]
name = "matte"
version = "0.1.0"
edition = "2021"

[dependencies]
anstyle.workspace = true
bat.workspace = true
fmtastic.workspace = true
hostname.workspace = true
matte-language-names = { path = "../matte-language-names" }
pulldown-cmark.workspace = true
smallvec.workspace = true
trait-set.workspace = true
unicode-linebreak-chunked.workspace = true
unicode-width.workspace = true
url.workspace = true

[dev-dependencies]
insta = { workspace = true, features = ["glob"] }
serde = { workspace = true, features = ["derive"] }
toml.workspace = true
