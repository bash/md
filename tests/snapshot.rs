use insta::{assert_snapshot, glob};
use matte::{render, supported_parser_options, Options};
use pulldown_cmark::Parser;
use std::fs::read_to_string;

#[test]
fn test_snippets() {
    glob!("snippets/*.md", |path| {
        let markdown = read_to_string(path).unwrap();
        let rendered = render_to_string(&markdown);
        assert_snapshot!(rendered);
    })
}

fn render_to_string(input: &str) -> String {
    let parser = Parser::new_ext(input, supported_parser_options());
    let mut buffer = Vec::new();
    let options = Options::plain_text(120);
    render(parser, &mut buffer, options).unwrap();
    String::from_utf8(buffer).unwrap()
}
