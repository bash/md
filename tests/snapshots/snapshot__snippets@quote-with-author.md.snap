---
source: tests/snapshot.rs
expression: rendered
input_file: tests/snippets/quote-with-author.md
---
â”ƒ[0m [0mThe famous pipe. How people reproached me for it! And yet, could you stuff my pipe? No, it's just a [0m
â”ƒ[0m [0mrepresentation, is it not? So if I had written on my picture "This is a pipe", I'd have been lying![0m
    â€• [0mRenÃ© Magritte[0m
